postgres-types/src/pg_lsn.rs

```diff
@@ -0,0 +1,79 @@
+//! Log Sequence Number (LSN) type for PostgreSQL Write-Ahead Log
+//! (WAL), also known as the transaction log.
+
+use bytes::BytesMut;
+use postgres_protocol::types;
+use std::error::Error;
+use std::fmt;
+use std::str::FromStr;
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+/// Postgres `PG_LSN` type.
+#[derive(Clone, Copy, Eq, Ord, PartialEq, PartialOrd)]
+pub struct PgLsn(u64);
+
+/// Error parsing LSN.
+#[derive(Debug)]
+pub struct ParseLsnError;
```

Comment:
Can you add a private field to this type (e.g. `pub struct ParseLsnError(());` so it can't be constructed externally?